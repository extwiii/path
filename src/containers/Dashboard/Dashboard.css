.Dashboard-section {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  background-color: #eee;
  flex-wrap: wrap;
  max-height: 512px;
  overflow-y: scroll;
  padding: 20px;
}

@media (min-width: 900px) {
  .Dashboard-section {
    max-width: 1024px;
    margin: 0px auto;
  }
}
