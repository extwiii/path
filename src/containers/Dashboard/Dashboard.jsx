import React from 'react';
import DashboardCard from '../../components/DashboardCard/DashboardCard';
import './Dashboard.css';

const dataComesFromServer = [
  { id: 1, title: 'GDPR', description: 'Make sure your business is compliant' },
  {
    id: 2,
    title: 'Managing people',
    description: 'Solve your everyday HR stuff with easy-to-follow processes',
  },
  {
    id: 3,
    title: 'Hiring people',
    description:
      'Build the right team with a smooth hiring process or check your business has all it needs to comply with employment law',
  },
];

class Dashboard extends React.Component {
  state = {
    selectedTopic: '',
  };

  onClickHandler = (selectedTopic) => {
    this.setState({ selectedTopic });
  };

  render() {
    const { selectedTopic } = this.state;
    const getData = dataComesFromServer.map(item => (
      <DashboardCard
        key={item.id}
        title={item.title}
        description={item.description}
        onClick={this.onClickHandler}
      />
    ));
    return (
      <div className="Dashboard-section">
        {getData}
        {selectedTopic}
      </div>
    );
  }
}

export default Dashboard;
