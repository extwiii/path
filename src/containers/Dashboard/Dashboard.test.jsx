import React from 'react';
import { shallow } from 'enzyme';

import Dashboard from './Dashboard';

describe('<Dashboard />', () => {
  it('renders without crash', () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onClickhandler and updates state', () => {
    const wrapper = shallow(<Dashboard />);
    wrapper.instance().onClickHandler('Test');
    expect(wrapper.state('selectedTopic')).toEqual('Test');
  });
});
