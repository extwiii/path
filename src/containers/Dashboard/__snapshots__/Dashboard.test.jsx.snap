// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <DashboardCard
            description="Make sure your business is compliant"
            onClick={[Function]}
            title="GDPR"
          />,
          <DashboardCard
            description="Solve your everyday HR stuff with easy-to-follow processes"
            onClick={[Function]}
            title="Managing people"
          />,
          <DashboardCard
            description="Build the right team with a smooth hiring process or check your business has all it needs to comply with employment law"
            onClick={[Function]}
            title="Hiring people"
          />,
        ],
        "",
      ],
      "className": "Dashboard-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "description": "Make sure your business is compliant",
          "onClick": [Function],
          "title": "GDPR",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "description": "Solve your everyday HR stuff with easy-to-follow processes",
          "onClick": [Function],
          "title": "Managing people",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "description": "Build the right team with a smooth hiring process or check your business has all it needs to comply with employment law",
          "onClick": [Function],
          "title": "Hiring people",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <DashboardCard
              description="Make sure your business is compliant"
              onClick={[Function]}
              title="GDPR"
            />,
            <DashboardCard
              description="Solve your everyday HR stuff with easy-to-follow processes"
              onClick={[Function]}
              title="Managing people"
            />,
            <DashboardCard
              description="Build the right team with a smooth hiring process or check your business has all it needs to comply with employment law"
              onClick={[Function]}
              title="Hiring people"
            />,
          ],
          "",
        ],
        "className": "Dashboard-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "description": "Make sure your business is compliant",
            "onClick": [Function],
            "title": "GDPR",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "description": "Solve your everyday HR stuff with easy-to-follow processes",
            "onClick": [Function],
            "title": "Managing people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "description": "Build the right team with a smooth hiring process or check your business has all it needs to comply with employment law",
            "onClick": [Function],
            "title": "Hiring people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
